simulation/test_torrent_status.cpp

```diff
@@ -41,62 +41,139 @@ using namespace sim;
 namespace lt = libtorrent;
 
 // this is a test for torrent_status time counters are correct
+// it is a long running test, we did already try to improve it:
+// - less slow peers vs many fast peers
+// - slow peers with smal torrent vs fast peers with large torrent
+// - less peers with large torrents vs more peers with smal torrent
+// - connection limit vs parallel requests
+// as conlusion the following combination take the least amount of
+// execution time per tick:
+// - many slow peers with small torrent and connection limit
 TORRENT_TEST(status_timers)
 {
-	lt::time_point start_time;
+	lt::time_point startseed_time;
+	lt::time_point last_active_time;
 	lt::torrent_handle handle;
-
-	setup_swarm(1, swarm_test::upload
+	int last_tick;
+	// at least 11s intervall because after resume it takes 11s untill we unchuck
+	// new pieces at this slow upload rate
+	int ticks_check_intervall = 20;
+	seconds active_time = seconds(0);
+	bool tick_is_in_active_range = false;
+
+	setup_swarm(10, swarm_test::upload_slow
```

Comment:
could you explain again why you need 10 nodes? I would expect it to be 10 times as expensive to simulate 10 sessions instead of one. as far as I can tell, all you want is the smallest amount of upload or download activity (to set the timestamps) and then just wait. during the wait, presumably you want to do as little work as possible. e.g. as few session instances as possible.

I don't understand why you would need a larger torrent. All you need to set the timestamps is a single byte, no?